js/core.js

```diff
@@ -619,11 +626,73 @@ class ExtensionResources {
     }
 }
 
+class ContextMenu {
+    static async init() {
+        ContextMenu.updateContextMenu();
+        chrome.contextMenus.onClicked.addListener(ContextMenu.genericOnClick);
+        chrome.storage.onChanged.addListener(ContextMenu.updateContextMenu);
+    }
+
+    static genericOnClick(info, tab) {
+        console.log("item " + info.menuItemId + " was clicked");
+        console.log("info: " + JSON.stringify(info));
+        console.log("tab: " + JSON.stringify(tab));
+    
+        switch (info.menuItemId) {
+            case "context_steam_store":
+                chrome.tabs.create({url: "https://store.steampowered.com/search/?term=" + encodeURIComponent(info.selectionText)});
+                break;
+            case "context_steam_market":
+                chrome.tabs.create({url: "https://steamcommunity.com/market/search?q=" + encodeURIComponent(info.selectionText)});
+                break;
+            case "context_itad":
+                chrome.tabs.create({url: "https://isthereanydeal.com/#/filter:&search/" + encodeURIComponent(info.selectionText) + ";/scroll:%23gamelist"});
+                break;
+            case "context_steamdb":
+                chrome.tabs.create({url: "https://steamdb.info/search/?q=" + encodeURIComponent(info.selectionText)});
+                break;
+            case "context_steamdb_instant":
+                chrome.tabs.create({url: "https://steamdb.info/instantsearch/?idx=steamdb&q=" + encodeURIComponent(info.selectionText)});
+                break;
+            case "es_context_steamkeys":
+                let steamkeys = info.selectionText.match(/[A-NP-RTV-Z02-9]{5}(-[A-NP-RTV-Z02-9]{5}){2}/g);
+                if (!steamkeys || steamkeys.length === 0) {
+                    window.alert("No keys found!"); // doesn't work??
+                    return;
+                }
+
+                let param = steamkeys.length === 1 ? "key=" : "keys=";
+                chrome.tabs.create({url: "https://store.steampowered.com/account/registerkey?" + param + encodeURIComponent(steamkeys.join())});
+                break;
+        }
+    }
+    
+    static buildContextMenu() {
+        let options = ["context_steam_store", "context_steam_market", "context_itad", "context_steamdb", "context_steamdb_instant", "context_steam_keys"];
+        for (let option of options) {
+            if (!SyncedStorage.get(option)) { return; }
+            console.log("ADDING " + option);
+            chrome.contextMenus.create({
+                "id": option,
```

Comment:
For some reason this `id` property is required to work, while it shouldn't be required at all...

Reply:
`Extensions using event pages must pass an id parameter to chrome.contextMenus.create`